extension/js/content_scripts/webmail/gmail_element_replacer.ts

```diff
@@ -550,4 +557,16 @@ export class GmailElementReplacer implements WebmailElementReplacer {
     }
   }
 
+  private addOrRemoveEndSessionBtnIfNeeded = async () => {
```

Comment:
This logic is exactly the same as in inbox. You can create a file `js/common/webmail.ts` where we can put shared functionality, for example.

Reply:
Yes, it will be better,

Thanks!